### PR TITLE
Utilize go generate

### PR DESCRIPTION
**Pull request by marquiz**
Use 'go generate' for auto-generating code. Drop the old 'mock' and
'apigen' makefile targets. Those are replaced with a single
  make generate

which (re-)generates everything.

### CODE REVIEW

File: pkg/apihelper/apihelpers.go [hunk 1]
```diff
@@ -16,6 +16,8 @@ limitations under the License.
 
 package apihelper
 
+//go:generate mockery --name=APIHelpers --inpkg
```
Comment (zvonkok) [inline pkg/apihelper/apihelpers.go:19]: Leftover?

**Review by ArangoGutierrez — CHANGES_REQUESTED** [review]
Just  a nit/question, should we add this new make target to `https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/scripts/prepare-release.sh` ? so all is freshly generated when cutting a release (if not, this lgtm)

**Comment by marquiz** [comment]
> Just a nit/question, should we add this new make target to `https://github.com/kubernetes-sigs/node-feature-discovery/blob/master/scripts/prepare-release.sh` ? so all is freshly generated when cutting a release (if not, this lgtm)

I wouldn't add it there – the auto-generated parts should be aligned before cutting the release. I've been thinking about adding a verify step, though. To check that everything's in sync. However, we're not quite ready for that yet. We have no nice way of ensuring that we use the correct version of the auto-generation tools. Thus, all of this will be left as a future enhancement.

**Review by ArangoGutierrez — APPROVED** [review]
My concern has been addressed 
/lgtm

**Pull request merged**